added _deletedAt to $deletedAt translation

shimonewman opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Added support for handling a deleted timestamp attribute in the database adapter.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shimonewman committed:
added _deletedAt to $deletedAt translation

abnegate commented on src/Database/Adapter/SQL.php:
```diff
@@ -1738,6 +1738,7 @@ protected function getInternalKeyForAttribute(string $attribute): string
             '$tenant' => '_tenant',
             '$createdAt' => '_createdAt',
             '$updatedAt' => '_updatedAt',
+            '$deletedAt' => '_deletedAt',
```
@copilot Address this in this PR

Copilot committed:
add symmetric _deletedAt handling in getDocument and getAttributeProjection

Agent-Logs-Url: https://github.com/utopia-php/database/sessions/68d0e3c1-0a6e-486c-875a-24e50241ccf4

Co-authored-by: abnegate <5857008+abnegate@users.noreply.github.com>

abnegate committed:
Merge branch 'main' into feat-_deletedAt-attr

abnegate commented:
@copilot Address greptile comments

Copilot committed:
fix: only select _deletedAt when explicitly requested; add Mongo adapter support

Agent-Logs-Url: https://github.com/utopia-php/database/sessions/b90d2aa4-7e7a-4baf-9931-9b151266a79e

Co-authored-by: abnegate <5857008+abnegate@users.noreply.github.com>
